yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -491,21 +533,66 @@ func (yb *TargetYugabyteDB) importBatch(conn *pgx.Conn, batch Batch, args *Impor
 		}
 	}()
 
-	// Check if the split is already imported.
-	var alreadyImported bool
-	alreadyImported, rowsAffected, err = yb.isBatchAlreadyImported(tx, batch)
+	// using the conn on which the transaction is executing which should ensure -
+	// all DB operations inside copyBatchCore will be a part of the transaction
+	rowsAffected, err = yb.copyBatchCore(tx.Conn(), batch, args)
+	return rowsAffected, err
+}
+
+func (yb *TargetYugabyteDB) importBatchFast(conn *pgx.Conn, batch Batch, args *ImportBatchArgs) (int64, error) {
+	log.Infof("importing %q using COPY command fast path(non transactional)", batch.GetFilePath())
+	// running copy without transaction
+	rowsAffected, err := yb.copyBatchCore(conn, batch, args)
+
+	/*
+		Check if the error is violates unique constraint then use the fast path recovery to ingest this batch
+		Case: If the table already has the data(before import started) which is being imported
+
+		Lets say the importBatchFastRecover fails with some trasient DB error. In that case,
+		caller(fileTaskImporter.importBatch() function) takes care of retrying with importBatchFastRecover
+	*/
+	if err != nil && strings.Contains(err.Error(), VIOLATES_UNIQUE_CONSTRAINT_ERROR) {
```

Comment:
The error that we are checking seems very generic. We should only retry if it's a PK violation, not any unique constraint error. Can we limit it only to PK violations?

Reply:
let's address this in the follow-up PR